frontend/server/src/Controllers/Certificate.php

```diff
@@ -385,6 +385,96 @@ private static function getCoderOfTheMonthCertificate(
         );
     }
 
+    /**
+     * Creates a Clarification for a contest or an assignment of a course
+     *
+     * @throws \OmegaUp\Exceptions\NotFoundException
+     *
+     * @return array{status: string}
+     *
+     * @omegaup-request-param int|null $certificates_cutoff
+     * @omegaup-request-param int|null $contest_id
+     */
+    public static function apiGenerateContestCertificates(\OmegaUp\Request $r) {
+        \OmegaUp\Controllers\Controller::ensureNotInLockdown();
+
+        $r->ensureMainUserIdentity();
+
+        $contestID = $r->ensureInt('contest_id');
+
+        // obtain the contest
+        $contest = \OmegaUp\DAO\Contests::getByPK($contestID);
+
+        $problemset = \OmegaUp\DAO\Problemsets::getByPK(
+            $contest->problemset_id
+        );
+
+        if (is_null($contest)) {
+            throw new \OmegaUp\Exceptions\NotFoundException('contestNotFound');
+        }
+        //check if is a certificate generator
+        if (!\OmegaUp\Authorization::isCertificateGenerator($r->identity)) {
+            throw new \OmegaUp\Exceptions\ForbiddenAccessException();
+        }
+
+        if ($contest->certificates_status !== 'uninitiated' && $contest->certificates_status !== 'retryable_error') {
+            return ['status' => 'ok'];
+        }
+
+        $certificateCutoff = $r->ensureOptionalInt('certificates_cutoff');
+
+        // add certificates_cutoff value to the course
+        if (!is_null($certificateCutoff)) {
+            $contest->certificate_cutoff = $certificateCutoff;
+        }
+
+        // update contest with the new value
+        \OmegaUp\DAO\Contests::update($contest);
+
+        // get contest info
+        $contestExtraInformation = \OmegaUp\DAO\Contests::getByAliasWithExtraInformation(
+            $contest->alias
+        );
+
+        // set RabbitMQ client parameters
+        $routing_key = 'ContestQueue';
+        $exchange = 'certificates';
+
+        //connection to rabbitmq
+        $channel = \OmegaUp\RabbitMQConnection::getInstance()->channel();
+
+        $scoreboard = \OmegaUp\Controllers\Contest::getScoreboard(
+            $contest,
+            $problemset,
+            $r->identity,
+            $contestExtraInformation['scoreboard_url']
+        );
+
+        $ranking = array_map(
+            fn ($rank) => ['username' => $rank['username'] , 'place' => $rank['place']],
```

Comment:
Parece que `place` no siempre está definido, así que podrías mandarle null en esos casos?

```suggestion
            fn ($rank) => ['username' => $rank['username'] , 'place' => $rank['place'] ?? null],
```